worker/mutation.go

```diff
@@ -195,6 +200,19 @@ func runSchemaMutation(ctx context.Context, updates []*pb.SchemaUpdate, startTs
 	var wg sync.WaitGroup
 	wg.Add(1)
 	defer wg.Done()
+	// This throttle allows is used to limit the number of files which are opened simultaneously
+	// by badger while building indexes for predicates in background.
+	maxOpenFileLimit, err := QueryMaxOpenFiles()
+	if err != nil {
+		// Setting to default value on unix systems
+		maxOpenFileLimit = 1024
+	}
+	glog.Infof("Max open files limit: %d", maxOpenFileLimit)
+	// Badger opens around 7 files for indexing per predicate.
+	// The throttle limit is set to maxOpenFileLimit/8 to ensure that indexing does not throw
+	// "Too many open files" error.
+	throttle := y.NewThrottle(maxOpenFileLimit / 8)
```

Comment:
This won't work. Each badger instance could open X number of files. We have no way of knowing how many files would badger open.

If you set throttle to 1024/8 = 128, you are assuming each badger instance will open at max 8 files which is wrong. This wouldn't work.

This isn't easy to fix. A simpler thing to do would be to ask the user to increate the ulimit on the system.

Reply:
Had a discussion with @jarifibrahim today afternoon. He has agreed that the solution would work in case of an empty database. I have addressed Naman's comments and fixed tests. @jarifibrahim can you take a look at the revised PR.